source-map.bs

```diff
@@ -256,14 +364,166 @@ with four fields represent mapped code where a corresponding name does not exist
+  <dd>A [=decoded source=] or null.</dd>
+
+  <dt><dfn>originalLine</dfn></dt>
+  <dd>A non-negative integer or null.</dd>
+
+  <dt><dfn>originalColumn</dfn></dt>
+  <dd>A non-negative integer or null.</dd>
+
+  <dt><dfn>name</dfn></dt>
+  <dd>A [=string=] or null.</dd>
+</dl>
+
+
+To <dfn>decode source map mappings</dfn> given a [=string=]
+<dfn for="decode source map mappings">|mappings|</dfn>, a [=list=] of [=strings=]
+<dfn for="decode source map mappings">|names|</dfn>, and a [=list=] of [=decoded source|decoded
+sources=] <dfn for="decode source map mappings">|sources|</dfn>, run the following steps:
+1. If |mappings| is not an [=ASCII string=], throw an error.
+1. If |mappings| contains any [=code unit=] other than:
+    - U+002C (,) or U+003B (;);
+    - U+0030 (0) to U+0039 (9);
+    - U+0041 (A) to U+005A (Z);
+    - U+0061 (a) to U+007A (z);
+    - U+002B (+), U+002F (/)
+    <!-- TODO: Is = allowed? -->
+
+    NOTE: These are the valid [[base64]] characters (excluding the padding character `=`), together
+    with `,` and `;`.
+
+    then throw an error.
+1. Let |decodedMappings| be a new empty [=list=].
+1. Let |groups| be the result of [=strictly split|strictly splitting=] |mappings| on `;`.
+1. Let |generatedLine| be 0.
+1. While |generatedLine| is less than |groups|'s [=list/size=]:
+    1. If |groups|[|generatedLine|] is not the empty string, then:
+        1. Let |segments| be the result of [=strictly split|strictly splitting=]
+            |groups|[|generatedLine|] on `,`.
+        1. Let |generatedColumn| be 0.
+        1. Let |sourceIndex| be 0.
+        1. Let |originalLine| be 0.
+        1. Let |originalColumn| be 0.
+        1. Let |nameIndex| be 0.
+        1. [=For each=] |segment| in |segments|:
+            1. Let |position| be a [=position variable=] for |segment|, initially pointing at
+                |segment|'s start.
+            1. [=Decode a base64 VLQ=] from |segment| given |position| and let
+                |relativeGeneratedColumn| be the result.
+            1. If |relativeGeneratedColumn| is null, [=optionally report an error=] and continue
+                with the next iteration.
+            1. Increase |generatedColumn| by |relativeGeneratedColumn|. If the result is negative,
+                [=optionally report an error=] and continue with the next iteration.
+            1. Let |decodedMapping| be a new [=decoded mapping=] whose
+                [=decoded mapping/generatedLine=] is |generatedLine|,
+                [=decoded mapping/generatedColumn=] is |generatedColumn|,
+                [=decoded mapping/originalSource=] is null,
+                [=decoded mapping/originalLine=] is null,
+                [=decoded mapping/originalColumn=] is null,
+                and [=decoded mapping/name=] is null.
+            1. Append |decodedMapping| to |decodedMappings|.
+            1. [=Decode a base64 VLQ=] from |segment| given |position| and let |relativeSourceIndex|
```

Comment:
`segment[position]` cannot be `,`/`;`, because segment is obtained by first splitting the string on those two chars. I'll handle the end-of-string case.